src/Microsoft.DotNet.Build.Tasks.Feed/PushToBlobFeed.cs

```diff
@@ -51,13 +63,22 @@ public override bool Execute()
             return !Log.HasLoggedErrors;
         }
 
-        private List<string> ConvertToStringLists(ITaskItem[] taskItems)
+        private List<string> ConvertToStringLists(ITaskItem[] taskItems, bool isNugetPackage)
         {
             List<string> stringList = new List<string>();
             foreach (var item in taskItems)
             {
-                stringList.Add(item.ItemSpec);
+                string fileSpec = item.ItemSpec;
+                string extension = Path.GetExtension(fileSpec);
+
+                // If packages is set to true it will only add files with ".nupkg" extension to the list (true == true)
+                // else this will add non-nupkg files to the list (false == false)
+                if ((extension == ".nupkg") == isNugetPackage)
+                {
+                    stringList.Add(item.ItemSpec);
```

Comment:
You should also support a relative path passed in as item metadata for the non-nuget packages.

Reply:
I think we are going to want to support something like https://github.com/dotnet/buildtools/blob/78c57e11af6a055c734e180ae3c40a5967429be6/src/Microsoft.DotNet.Build.CloudTestTasks/PackageFiles/PublishContent.targets#L24 to ensure we preserve the same directory structure when we upload.